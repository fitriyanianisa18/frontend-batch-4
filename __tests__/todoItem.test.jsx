import {render, screen, fireEvent} from '@testing-library/react'
import TodoItem from '@/components/todoComponents/todoItem'

const mockTodo = {
    id: 1,
    text: "Learn Testing",
    done: false
}

const mockIndex = 1

const toggleDone = jest.fn()
const toggleDelete = jest.fn()

describe("Todo Item", () => {
    it("render todo text", () => {
        render(<TodoItem index={mockIndex} todo={mockTodo} onDelete={toggleDelete} onDone={toggleDone}/>)

        const todoText = screen.getByText(/learn testing/i)
        expect(todoText).toBeInTheDocument()
        expect(todoText).toHaveTextContent("Learn Testing")
    })

    it("Call toggle delete function", () => {
        render(<TodoItem index={mockIndex} todo={mockTodo} onDelete={toggleDelete} onDone={toggleDone} />);

        const deleteButton = screen.getByText(/Ã—/i);
        fireEvent.click(deleteButton)

        expect(toggleDelete).toHaveBeenCalledWith(1)
    })

    it("Match snapshot", () => {
        const {asFragment} = render(<TodoItem index={mockIndex} todo={mockTodo} onDelete={toggleDelete} onDone={toggleDone} />);

        expect(asFragment()).toMatchSnapshot()
    })
})