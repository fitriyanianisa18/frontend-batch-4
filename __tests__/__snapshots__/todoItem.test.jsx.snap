// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Todo Item Match snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex items-center justify-between bg-white rounded p-3 shadow-sm"
  >
    <span
      class="px-4 py-2 rounded cursor-pointer bg-white hover:bg-gray-50"
    >
      1. Learn Testing
    </span>
    <button
      class="ml-3 text-red-400 hover: text-red-700 font-bold"
    >
      Ã—
    </button>
  </div>
</DocumentFragment>
`;
